expat/lib/xmlparse.c

```diff
@@ -8392,9 +8393,9 @@ getDebugLevel(const char *variableName, unsigned long defaultDebugLevel) {
   const char *const value = valueOrNull;
 
   errno = 0;
-  char *afterValue = (char *)value;
+  char *afterValue = NULL;
   unsigned long debugLevel = strtoul(value, &afterValue, 10);
-  if ((errno != 0) || (afterValue[0] != '\0')) {
+  if ((errno != 0) || (afterValue == value) || (afterValue[0] != '\0')) {
```

Comment:
@seanm I believe this line fixes a true (but masked) bug — arguably unexpected behavior — in function `getDebugLevel`. When the environment variable is the empty string, one may expect that the default is return (whatever its value) rather than a return of zero.  All 3 current uses in Expat match pattern `getDebugLevel("VARNAME", 0)` and that is masking the issue in practice, and that only.

My vote for making that fix a dedicated pull request.  What do you think?

For proof and/or reproduction:

```console
# gcc -std=c99 -Wextra -pedantic main.c && LEVEL= ./a.out
Variable ${LEVEL} got us level 0, default is level 123.

# cat main.c 
#include <errno.h>
#include <stdio.h>
#include <stdlib.h>

static unsigned long getDebugLevel(const char *variableName,
                                   unsigned long defaultDebugLevel) {
  const char *const valueOrNull = getenv(variableName);
  if (valueOrNull == NULL) {
    return defaultDebugLevel;
  }
  const char *const value = valueOrNull;

  errno = 0;
  char *afterValue = (char *)value;
  unsigned long debugLevel = strtoul(value, &afterValue, 10);
  if ((errno != 0) || (afterValue[0] != '\0')) {
    errno = 0;
    return defaultDebugLevel;
  }

  return debugLevel;
}

int main() {
  const unsigned long defaultLevel = 123;
  printf("Variable ${LEVEL} got us level %lu, default is level %lu.\n",
         getDebugLevel("LEVEL", defaultLevel), defaultLevel);
  return 0;
}
```

Reply:
>My vote for making that fix a dedicated pull request. What do you think?

Sure, if you want. Makes no difference to me. :)

Reply:
@seanm excellent, thank you!